src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useEffect} from 'react';
+import type {ComponentType} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import useNetwork from '@hooks/useNetwork';
+import {openPolicyAccountingPage} from '@libs/actions/PolicyConnections';
+import ONYXKEYS from '@src/ONYXKEYS';
+import withPolicy from './withPolicy';
+import type {WithPolicyProps} from './withPolicy';
+
+type WithPolicyConnectionsProps = WithPolicyProps;
+
+/**
+ * Higher-order component that fetches the connections data and populates
+ * the corresponding field of the policy object if the field is empty. It then passes the policy object
+ * to the wrapped component.
+ *
+ * Use this HOC when you need the policy object with its connections field populated.
+ *
+ * Only the active policy gets the complete policy data upon app start that includes the connections data.
+ * For other policies, the connections data needs to be fetched when it's needed.
+ */
+function withPolicyConnections(WrappedComponent: ComponentType<WithPolicyConnectionsProps>) {
+    function WithPolicyConnections({policy, policyDraft, route}: WithPolicyConnectionsProps) {
+        const {isOffline} = useNetwork();
+        const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${policy?.id ?? ''}` as const, {
+            initWithStoredValues: false,
+        });
+
+        useEffect(() => {
+            // When the accounting feature is not enabled, or if the connections data already exists,
+            // there is no need to fetch the connections data.
+            if (!policy || !policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!policy.connections) {
+                return;
+            }
+
+            openPolicyAccountingPage(policy.id);
+        }, [hasConnectionsDataBeenFetched, policy]);
+
+        if (status === 'loading' || !hasConnectionsDataBeenFetched) {
```

Comment:
This will always show the loader when the policy hasn't enabled the accounting feature. Should we let the `OpenPolicyAccountingPage` API be called even though the policy hasn't enabled the feature?

Reply:
Just to confirm, how is this accessible in the UI if accounting isn't enabled?

Reply:
@mollfpr, this page won't be acceesible for users who hasn't enabled the accounting feature.

Also, even if the user somehow be able to access this page, it won't cause the infinite loading state. `hasConnectionsDataBeenFetched` will be true if the fetch is attempted once (even if there is no data returned from the API call)

Reply:
also, just to be clear, the `connections` field might still be empty even if the user has enabled the accounting feature if the user hasn't connected to any accounting software

Reply:
This seems broken. I'm getting infinite loader when I try to visit the page again

https://github.com/Expensify/App/assets/16493223/bf5a6e67-2d38-4d8e-a0b0-a185364602e9

Reply:
@s77rt similar problem with going offline and then offline: https://github.com/Expensify/App/pull/39132#discussion_r1569411481

@hayata-suenaga is working on it here https://github.com/Expensify/App/pull/40375